source/rust_verify/src/verifier.rs

```diff
@@ -394,30 +402,113 @@ impl Verifier {
             }
         }
         let mut invalidity = false;
+        let CommandsWithContextX { span, desc, commands, spinoff_z3: _ } = &*commands_with_context;
         if commands.len() > 0 {
             air_context.blank_line();
             air_context.comment(comment);
         }
-        for CommandsWithContextX { span, desc, commands } in commands.iter().map(|x| &**x) {
-            for command in commands.iter() {
-                let time0 = Instant::now();
-                let result_invalidity = self.check_result_validity(
-                    compiler,
-                    error_as,
-                    air_context,
-                    assign_map,
-                    snap_map,
-                    &command,
-                    &(span, desc),
-                );
-                invalidity = invalidity || result_invalidity;
-                let time1 = Instant::now();
-                self.time_air += time1 - time0;
-            }
+        for command in commands.iter() {
+            let time0 = Instant::now();
+            let result_invalidity = self.check_result_validity(
+                compiler,
+                error_as,
+                air_context,
+                assign_map,
+                snap_map,
+                &command,
+                &(span, desc),
+            );
+            invalidity = invalidity || result_invalidity;
+            let time1 = Instant::now();
+            self.time_air += time1 - time0;
         }
+
         invalidity
     }
 
+    fn new_air_context_with_module_context(
```

Comment:
This function copied-and-pasted a lot of code from `verify_module_outer`.  Could you consolidate these so that there is only one copy?

Reply:
I agree, I will try to restructure the codes. Thank you!